C4SGSvc/src/main/java/org/c4sg/controller/OrganizationController.java

```diff
@@ -1,88 +1,110 @@
+    public Map<String, Object> createOrganization(@RequestBody @Valid OrganizationDto organizationDto) {
+        System.out.println("**************Create**************");
+        Map<String, Object> responseData = null;
+        organizationDto.setLogo(getLogoUploadPath(organizationDto.getName()));
+        try {
+            OrganizationDto createdOrganization = organizationService.createOrganization(organizationDto);
+            responseData = Collections.synchronizedMap(new HashMap<>());
+            responseData.put("organization", createdOrganization);
+        } catch (Exception e) {
+            System.err.println(e);
+        }
+        return responseData;
     }
-    
+
     @CrossOrigin
-    @RequestMapping(value="/api/organization/update/{id}", method = RequestMethod.PUT)
-    public Map<String, Object> updateOrganization(@PathVariable("id") int id, @RequestBody @Valid OrganizationDto organizationDto){
-    	System.out.println("**************Update : id=" + organizationDto.getId() + "**************");
-    	Map<String, Object> responseData = null;
-    	try{
-    		OrganizationDto updatedOrganization = organizationService.updateOrganization(id, organizationDto);
-    		responseData = Collections.synchronizedMap(new HashMap<>());
-    		responseData.put("organization", updatedOrganization);
-    	}catch(Exception e){
-    		System.err.println(e);
-    	}
-    	return responseData;
+    @RequestMapping(value = "/api/organization/update/{id}", method = RequestMethod.PUT)
+    public Map<String, Object> updateOrganization(@PathVariable("id") int id, @RequestBody @Valid OrganizationDto organizationDto) {
+        System.out.println("**************Update : id=" + organizationDto.getId() + "**************");
+        Map<String, Object> responseData = null;
+        try {
+            OrganizationDto updatedOrganization = organizationService.updateOrganization(id, organizationDto);
+            responseData = Collections.synchronizedMap(new HashMap<>());
+            responseData.put("organization", updatedOrganization);
+        } catch (Exception e) {
+            System.err.println(e);
+        }
+        return responseData;
     }
-    
+
     @CrossOrigin
-    @RequestMapping(value="/api/organization/delete/{id}", method = RequestMethod.DELETE)
-    public void deleteOrganization(@PathVariable("id") int id){
-    	System.out.println("************** Delete : id=" + id + "**************");
-    	
+    @RequestMapping(value = "/api/organization/delete/{id}", method = RequestMethod.DELETE)
+    public void deleteOrganization(@PathVariable("id") int id) {
+        System.out.println("************** Delete : id=" + id + "**************");
+
         try {
             organizationService.deleteOrganization(id);
         } catch (Exception e) {
             System.out.println(e);
         }
     }
+
+    private String getLogoUploadPath(String organizationName) {
```

Comment:
Lets move this method to the organization Service layer and would be great to encapsulate whole logic into createOrganization method instead of multiple calls from controller.